### ISSUE TITLE
Pydantic optional field being treated as required by pyright

### ISSUE THREAD

**Issue by mjperrone** [microsoft/pyright#7166]
**Describe the bug**

I have a [pydantic model](https://docs.pydantic.dev/latest/concepts/fields/) with an optional field. When I set the default value as `None` directly, there are no problem with pyright seeing that it is optional. If I use `Field` with a description set, pyright treats it as if it is actually a required field despite the type hint and default value set in pydantic.

```python
from typing import Optional
from pydantic import BaseModel, Field

class ModelWithOptionalFields(BaseModel):
    fieldWithNoDescription: Optional[str] = None
    fieldWithDescription: Optional[str] = Field(
        None,
        description="Optional field with a default value of None and a description",
    )
no_optional_values_set = ModelWithOptionalFields()
```

**VS Code extension or command-line**
`poetry 

** output **

```
  test.py:14:8 - error: Argument missing for parameter "fieldWithDescription" (reportGeneralTypeIssues)
```

**Versions**

```
pydantic = "^1.10.7"
pyright = "1.1.317"
```

**Comment by erictraut** [comment]
First, if you're really using pyright 1.1.317, please upgrade. The latest version is 1.1.349, so you're 32 versions out of date.

Second, pyright is working as designed in this case, so I don't consider it a bug.

Pyright has no knowledge about pyduantic or its specific behaviors. It's simply applying the rules based on the type information it's provided. In this case, pydantic is leveraging the `dataclass_transform` mechanism introduced in [PEP 681](https://peps.python.org/pep-0681/). It allows libraries that act like the stdlib `dataclass` to adopt the same behaviors with respect to type checking. However, pydantic's behavior does differ from `dataclass` in a few small ways. 

In your example, you're passing `None` to the `Field` constructor as a positional argument. This isn't allowed in `dataclass`. In a `dataclass` field, you need to specify a default using the keyword argument `default`. This is also supported by pydantic, so you can easily fix this in your code by making this small change.

```python
    fieldWithDescription: Optional[str] = Field(default=None)
```

**Comment by mjperrone** [comment]
Thank you for your quick and comprehensive response. I appreciate the pointer to PEP 681 so I can learn more about that.

For your greater context, the code I provided I did not write, instead it was generated from an OpenAPI spec using https://github.com/koxudaxi/datamodel-code-generator. I'll advocate for the generation to use the keyword argument field.

**Comment by imaurer** [comment]
Amazing. Working with zed and pyright for the first time and I was mad at pyright for yelling at me for my perfectly valid Pydantic code. And here is the answer. I added the defaults and I actually think the code is better now too. 🥳
